specs/protocol/superchain-configuration.md

```diff
@@ -64,3 +67,55 @@ When the Pause is activated, the following methods are disabled:
 1. `StandardBridge.finalizeBridgeERC20()`
 1. `StandardBridge.finalizeBridgeETH()`
 1. `L1ERC721Bridge.finalizeBridgeERC721()`
+
+## Dependency manager
+
+The `SuperchainConfig` contract will manage and keep track of the dependency graph.
+It will be queried as the source of truth to get which chains are part of the Superchain.
+It will also allow to add a chain to the op-governed cluster and update each chain’s dependency set.
+
+### Interface and properties
+
+The contract will add the following storage layout and function:
+
+**`SHARED_LOCKBOX`**
+
+- An immutable address pointing to the `SharedLockbox` contract.
+- This address MUST be immutable because there's only one `SharedLockbox` for each cluster.
+
+**`systemConfigs`**
+
+- A mapping that associates chain IDs with their respective SystemConfig addresses.
+- It will be used when updating dependencies along each chain.
+
+**`dependencySet`**
+
+- An `EnumerableSet` that stores the current list of chain IDs in the dependency set.
```

Comment:
There isn't context on what an `EnumerableSet` is. Its an implementation detail, we should describe the properties of this data structure that we depend on